actors/systemsettings/src/main/java/org/sunbird/systemsettings/actors/SystemSettingsActor.java

```diff
@@ -83,33 +81,23 @@ private void getAllSystemSettings() {
     sender().tell(response, self());
   }
 
-  @SuppressWarnings("unchecked")
   private void setSystemSetting(Request actorMessage) {
     ProjectLogger.log("SystemSettingsActor: setSystemSetting called", LoggerEnum.DEBUG.name());
 
     Map<String, Object> request = actorMessage.getRequest();
-    
-    Boolean isPhoneUnique = ((String) request.get(JsonKey.FIELD)).equalsIgnoreCase(JsonKey.PHONE_UNIQUE) ? true : false;
-    Boolean isEmailUnique = ((String) request.get(JsonKey.FIELD)).equalsIgnoreCase(JsonKey.EMAIL_UNIQUE) ? true : false;
-    
-    if (isPhoneUnique || isEmailUnique) {
-      SystemSetting systemSetting = systemSettingDaoImpl.readByField((String)request.get(JsonKey.FIELD));
-
-      Boolean existingValue = Boolean.parseBoolean(systemSetting.getValue());
-      Boolean requiredValue = Boolean.parseBoolean((String) request.get(JsonKey.VALUE));
-
-      if (existingValue == false && requiredValue == true) { 
-        if (isPhoneUnique) {
-          userClient.esVerifyPhoneUniqueness();
-        } else {
-          userClient.esVerifyEmailUniqueness();
-        }
-      }
+    String id = (String) request.get(JsonKey.ID);
+    String field = (String) request.get(JsonKey.FIELD);
+    if (JsonKey.PHONE_UNIQUE.equalsIgnoreCase(field)
+        || JsonKey.EMAIL_UNIQUE.equalsIgnoreCase(field)
+        || JsonKey.PHONE_UNIQUE.equalsIgnoreCase(id)
```

Comment:
actually check on id is required because this is the primary key on which update will occur and in system setting table id and field both have same value as of now, so at safer side i put check on both